state-chain/runtime/src/lib.rs

```diff
@@ -1342,6 +1344,53 @@ impl_runtime_apis! {
 				None
 			}
 		}
+
+		fn cf_witness_count(hash: pallet_cf_witnesser::CallHash) -> Option<FailingWitnessValidators> {
+			// helper function to compute the number of bits set and their indices
+			fn count_set_bits_and_indices(bytes: &[u8], size: usize) -> (u32, Vec<usize>) {
```

Comment:
I think I didn't express very well what I meant with my other comment. 

The witnesser pallet already uses a crate `bitvec` that has utilities to index into a vector of bits. 

You could load the votes into a BitVec and then just iterate this to get a Vec of (authority, vote, vanity_name). Something like this:

```rust
let votes = BitVec::from_vec(Votes::<Runtime>::get(callhash)?);
let vote_count = votes.count_ones();
let missing_vote_count = votes.count_zeros();

let result: Vec<_> = votes.iter()
	// by_vals is needed to convert to true/false bool values.
	.by_vals()
	// authorities are stored in the same order as the votes
	.zip(authorities)
	.map(|(vote, account_id)| (
		account_id,
		vote,
		vanity_names[account_id],
	))
	.collect();
```

This can all be implemented as a public fn inside the witnesser pallet.